File: 006/6.c
// What is the difference between the sum of the squares and the square of the sums?

#include <stdio.h>

int main () {

	int range = 100;
	int sum = 0; // Warum beginnt die sum sonst bei 1551883176?!
	int sumsq = 0;
	int diff;
	int i = 1;
	int helper;

	while (i < range + 1) {
		helper = i * i;
		sum += helper;
		i++;
	}
	printf("sum: %i\n", sum);

	i = 1;
	helper = 0;

	while (i < range + 1) {
		helper += i;
		i++;
	}
	sumsq = helper * helper;
	printf("sumsq: %i\n", sumsq);

	diff = sumsq - sum;
	printf("diff: %i\n", diff);

	return 0;
}

File: 006/6.swift
// The sum of the squares of the first ten natural numbers is,
// 1^2 + 2^2 + ... + 10^2 = 385
//
// The square of the sum of the first ten natural numbers is,
// (1 + 2 + ... + 10)^2 = 55^2 = 3025
//
// Hence the difference between the sum of the squares of the first ten 
// natural numbers and the square of the sum is 3025 − 385 = 2640.
//
// Find the difference between the sum of the squares of the first one
// hundred natural numbers and the square of the sum.

func sumOfSquares(number: Int) -> Int {
    var sum = 0
    for i in 1...number {
        sum += i * i
    }
    return sum
}

func squareOfSums(number: Int) -> Int {
    var sum = 0
    for i in 1...number {
        sum += i
    }
    return sum * sum
}

let range = 100
let difference = squareOfSums(range) - sumOfSquares(range)

println(difference)
